lightning/src/routing/router.rs

```diff
@@ -962,6 +980,43 @@ impl PaymentParameters {
 	}
 }
 
+/// Information manually provided by the user to route the payment
+#[derive(Clone, Copy)]
+pub struct ManualRoutingParameters {
+	/// Same as [`RouteParameters::max_total_routing_fee_msat`]
+	pub max_total_routing_fee_msat: Option<u64>,
+	/// Same as [`PaymentParameters::max_total_cltv_expiry_delta`]
+	pub max_total_cltv_expiry_delta: Option<u32>,
+	/// Same as [`PaymentParameters::max_path_count`]
+	pub max_path_count: Option<u8>,
+	/// Same as [`PaymentParameters::max_channel_saturation_power_of_half`]
+	pub max_channel_saturation_power_of_half: Option<u8>,
```

Comment:
I’ve made all the fields optional, so they act as overrides. If left as `None`, the values from `PaymentParameters` (created using the invoice and default settings) will be used. I’ve also updated the docs to make this clearer for this struct.

Would love to hear your feedback on this API design! Thanks so much!

Reply:
Expanded the role of struct in [pr3342.04](https://github.com/shaavan/rust-lightning/commits/pr3342.04), allowing configuring parameters to the user instead of simply overriding them.
Hence remove the unnecessary option, and use default if the value is absent.
Thanks for the pointer!